piper/buttonpage.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2017 Jente Hidskes <hjdskes@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+from gettext import gettext as _
+
+from .buttondialog import ButtonDialog
+from .mousemap import MouseMap
+from .optionbutton import OptionButton
+
+import gi
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk
+
+
+class ButtonPage(Gtk.Box):
+    """The second stack page, exposing the button configuration."""
+
+    __gtype_name__ = "ButtonPage"
+
+    def __init__(self, ratbagd_device, *args, **kwargs):
+        """Instantiates a new ButtonPage.
+
+        @param ratbag_device The ratbag device to configure, as
+                             ratbagd.RatbagdDevice
+        """
+        Gtk.Box.__init__(self, *args, **kwargs)
+        self._device = ratbagd_device
+        self._init_ui()
+
+    def _init_ui(self):
+        profile = self._device.active_profile
+
+        mousemap = MouseMap("#Buttons", self._device, spacing=20, border_width=20)
+        self.pack_start(mousemap, True, True, 0)
+
+        sizegroup = Gtk.SizeGroup(Gtk.SizeGroupMode.HORIZONTAL)
+        for ratbagd_button in profile.buttons:
+            index = ratbagd_button.index
+            # TODO: add the correct *mapping to the label
+            button = OptionButton(_("Button {}").format(index))
+            button.connect("clicked", self._on_button_clicked, ratbagd_button)
+            mousemap.add(button, "#button{}".format(index))
+            sizegroup.add_widget(button)
+
+    def _on_button_clicked(self, button, ratbagd_button):
+        # Presents the ButtonDialog to configure the mouse button corresponding
+        # to the clicked button.
+        profile = self._device.active_profile
+        n_buttons = len(profile.buttons)
```

Comment:
This part in particular I am not happy with, but it is the simplest approach for now. I could pass the complete profile instead of only the number of buttons, but I'm not sure how that will change once I add support for profiles. Although, similarly, I am not sure how this will change either -- I only know that this *will* change once I start on profiles.